# Sales Data Exploratory Analysis


## 1. Data Understanding



In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
rawsales = pd.read_csv('salesdata.csv')

I´ve successfully loaded the csv into the dataframe variable (rawsales)

In [3]:
print(rawsales)

     Order ID  Amount  Profit  Quantity         Category      Sub-Category  \
0     B-26776    9726    1275         5      Electronics  Electronic Games   
1     B-26776    9726    1275         5      Electronics  Electronic Games   
2     B-26776    9726    1275         5      Electronics  Electronic Games   
3     B-26776    4975    1330        14      Electronics          Printers   
4     B-26776    4975    1330        14      Electronics          Printers   
...       ...     ...     ...       ...              ...               ...   
1189  B-26370    8825    3594        15        Furniture            Tables   
1190  B-26298    2082     642         8      Electronics            Phones   
1191  B-26298    2082     642         8      Electronics            Phones   
1192  B-26298    2082     642         8      Electronics            Phones   
1193  B-25068     914     163        13  Office Supplies           Markers   

     PaymentMode  Order Date        CustomerName     State     

In [4]:
rawsales.head()

,Order ID,Amount,Profit,Quantity,Category,Sub-Category,PaymentMode,Order Date,CustomerName,State,City,Year-Month
0,B-26776,9726,1275,5,Electronics,Electronic Games,UPI,2023-06-27,David Padilla,Florida,Miami,2023-06
1,B-26776,9726,1275,5,Electronics,Electronic Games,UPI,2024-12-27,Connor Morgan,Illinois,Chicago,2024-12
2,B-26776,9726,1275,5,Electronics,Electronic Games,UPI,2021-07-25,Robert Stone,New York,Buffalo,2021-07
3,B-26776,4975,1330,14,Electronics,Printers,UPI,2023-06-27,David Padilla,Florida,Miami,2023-06
4,B-26776,4975,1330,14,Electronics,Printers,UPI,2024-12-27,Connor Morgan,Illinois,Chicago,2024-12


In [5]:
rawsales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1194 entries, 0 to 1193
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Order ID      1194 non-null   object
 1   Amount        1194 non-null   int64 
 2   Profit        1194 non-null   int64 
 3   Quantity      1194 non-null   int64 
 4   Category      1194 non-null   object
 5   Sub-Category  1194 non-null   object
 6   PaymentMode   1194 non-null   object
 7   Order Date    1194 non-null   object
 8   CustomerName  1194 non-null   object
 9   State         1194 non-null   object
 10  City          1194 non-null   object
 11  Year-Month    1194 non-null   object
dtypes: int64(3), object(9)
memory usage: 112.1+ KB


The dataset contains 1,194 observations across 12 variables. There are no immediately apparent null values. Three variables are numeric, while nine are stored as objects. Order Date is currently represented as an object rather than a datetime variable and may require conversion before time-series analysis.

In [6]:
rawsales.shape

(1194, 12)

Rows = 1194 
Columns = 12

In [7]:
rawsales.head(10)

,Order ID,Amount,Profit,Quantity,Category,Sub-Category,PaymentMode,Order Date,CustomerName,State,City,Year-Month
0,B-26776,9726,1275,5,Electronics,Electronic Games,UPI,2023-06-27,David Padilla,Florida,Miami,2023-06
1,B-26776,9726,1275,5,Electronics,Electronic Games,UPI,2024-12-27,Connor Morgan,Illinois,Chicago,2024-12
2,B-26776,9726,1275,5,Electronics,Electronic Games,UPI,2021-07-25,Robert Stone,New York,Buffalo,2021-07
3,B-26776,4975,1330,14,Electronics,Printers,UPI,2023-06-27,David Padilla,Florida,Miami,2023-06
4,B-26776,4975,1330,14,Electronics,Printers,UPI,2024-12-27,Connor Morgan,Illinois,Chicago,2024-12
5,B-26776,4975,1330,14,Electronics,Printers,UPI,2021-07-25,Robert Stone,New York,Buffalo,2021-07
6,B-26942,1525,185,12,Office Supplies,Pens,Debit Card,2024-05-11,John Fields,Florida,Orlando,2024-05
7,B-26942,1525,185,12,Office Supplies,Pens,Debit Card,2021-10-09,Clayton Smith,Florida,Miami,2021-10
8,B-26640,883,117,10,Electronics,Laptops,EMI,2022-11-18,Richard Kelley,California,Los Angeles,2022-11
9,B-26640,8127,3551,16,Furniture,Tables,Credit Card,2022-11-18,Richard Kelley,California,Los Angeles,2022-11


This shows the first 10 entries in the dataset

In [8]:
rawsales.describe()

,Amount,Profit,Quantity
count,1194.000000,1194.000000,1194.000000
mean,5178.089615,1348.992462,10.674204
std,2804.921955,1117.992573,5.777102
min,508.000000,50.000000,1.000000
25%,2799.000000,410.000000,6.000000
50%,5152.000000,1014.000000,11.000000
75%,7626.000000,2035.000000,16.000000
max,9992.000000,4930.000000,20.000000


- The dataset contains 1194 observations across all 3 numerical variables, .info() shows that there are no missing values recognized by pandas in these column.

- describe() has made profits distribution worth investigating

- The middle 50% of Amount values fall between 2,799 and 7,626 as shown by the 25th and 75th percentiles.

- Profit has a wide and and considerable range from 50 to 4930.

In [9]:
rawsales.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
1189    False
1190    False
1191    False
1192    False
1193    False
Length: 1194, dtype: bool

In [10]:
rawsales.duplicated().sum()

np.int64(0)

No exact duplicate rows were identified in the dataset

In [11]:
rawsales.isnull()

,Order ID,Amount,Profit,Quantity,Category,Sub-Category,PaymentMode,Order Date,CustomerName,State,City,Year-Month
0,False,False,False,False,False,False,False,False,False,False,False,False
1,False,False,False,False,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,False,False,False,False,False,False
3,False,False,False,False,False,False,False,False,False,False,False,False
4,False,False,False,False,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...
1189,False,False,False,False,False,False,False,False,False,False,False,False
1190,False,False,False,False,False,False,False,False,False,False,False,False
1191,False,False,False,False,False,False,False,False,False,False,False,False
1192,False,False,False,False,False,False,False,False,False,False,False,False


In [12]:
rawsales.isnull().sum()

Order ID        0
Amount          0
Profit          0
Quantity        0
Category        0
Sub-Category    0
PaymentMode     0
Order Date      0
CustomerName    0
State           0
City            0
Year-Month      0
dtype: int64

0 null values across all 12 columns were detected

In [13]:
rawsales["Category"].unique()

array(['Electronics', 'Office Supplies', 'Furniture'], dtype=object)

This shows what the unique values are in the column.

In [14]:
rawsales["Category"].nunique()

3

This tells us how many unique values there are in the column.

In [15]:
rawsales["Category"].value_counts()

Category
Furniture          407
Office Supplies    399
Electronics        388
Name: count, dtype: int64

In [16]:
rawsales["PaymentMode"].unique()

array(['UPI', 'Debit Card', 'EMI', 'Credit Card', 'COD'], dtype=object)

In [17]:
rawsales["PaymentMode"].value_counts()

PaymentMode
Debit Card     260
Credit Card    258
UPI            252
EMI            218
COD            206
Name: count, dtype: int64

There are 5 payments methods.

Debit card was the most frequent with 260 records.

In [18]:
rawsales["State"].unique()

array(['Florida', 'Illinois', 'New York', 'California', 'Texas', 'Ohio'],
      dtype=object)

In [19]:
rawsales["State"].value_counts()

State
New York      226
California    218
Florida       200
Texas         189
Illinois      181
Ohio          180
Name: count, dtype: int64

In [20]:

rawsales["PaymentMode"].groupby(rawsales["State"]).value_counts()

State       PaymentMode
California  Credit Card    48
            UPI            47
            Debit Card     45
            EMI            41
            COD            37
Florida     UPI            46
            Credit Card    43
            COD            42
            Debit Card     39
            EMI            30
Illinois    UPI            45
            EMI            39
            Credit Card    38
            Debit Card     31
            COD            28
New York    Debit Card     57
            UPI            53
            Credit Card    47
            COD            35
            EMI            34
Ohio        Debit Card     42
            Credit Card    39
            EMI            36
            UPI            33
            COD            30
Texas       Debit Card     46
            Credit Card    43
            EMI            38
            COD            34
            UPI            28
Name: count, dtype: int64

In [21]:
pd.crosstab(rawsales["State"], rawsales["PaymentMode"])

PaymentMode,COD,Credit Card,Debit Card,EMI,UPI
State,,,,,
California,37,48,45,41,47
Florida,42,43,39,30,46
Illinois,28,38,31,39,45
New York,35,47,57,34,53
Ohio,30,39,42,36,33
Texas,34,43,46,38,28


In [22]:
pd.crosstab(rawsales["State"], rawsales["PaymentMode"], margins=True, normalize="index")

PaymentMode,COD,Credit Card,Debit Card,EMI,UPI
State,,,,,
California,0.169725,0.220183,0.206422,0.188073,0.215596
Florida,0.210000,0.215000,0.195000,0.150000,0.230000
Illinois,0.154696,0.209945,0.171271,0.215470,0.248619
New York,0.154867,0.207965,0.252212,0.150442,0.234513
Ohio,0.166667,0.216667,0.233333,0.200000,0.183333
Texas,0.179894,0.227513,0.243386,0.201058,0.148148
All,0.172529,0.216080,0.217755,0.182580,0.211055


In [23]:
pd.crosstab(rawsales["State"], rawsales["PaymentMode"], margins=True, normalize="index") * 100

PaymentMode,COD,Credit Card,Debit Card,EMI,UPI
State,,,,,
California,16.972477,22.018349,20.642202,18.807339,21.559633
Florida,21.000000,21.500000,19.500000,15.000000,23.000000
Illinois,15.469613,20.994475,17.127072,21.546961,24.861878
New York,15.486726,20.796460,25.221239,15.044248,23.451327
Ohio,16.666667,21.666667,23.333333,20.000000,18.333333
Texas,17.989418,22.751323,24.338624,20.105820,14.814815
All,17.252931,21.608040,21.775544,18.257956,21.105528


In [24]:
pd.crosstab(rawsales["State"], rawsales["PaymentMode"], margins=True, normalize="index").mul(100).round(2)

PaymentMode,COD,Credit Card,Debit Card,EMI,UPI
State,,,,,
California,16.97,22.02,20.64,18.81,21.56
Florida,21.00,21.50,19.50,15.00,23.00
Illinois,15.47,20.99,17.13,21.55,24.86
New York,15.49,20.80,25.22,15.04,23.45
Ohio,16.67,21.67,23.33,20.00,18.33
Texas,17.99,22.75,24.34,20.11,14.81
All,17.25,21.61,21.78,18.26,21.11


Rounding up to 2 decimal point


Payment-method distribution varies somewhat between states. Florida has the highest proportion of COD transactions at 21%, while debit-card transactions are most prominent in New York (25.22%) and Texas (24.34%). UPI shows a notable difference between states, ranging from 14.81% in Texas to 24.86% in Illinois. In contrast, credit-card usage is comparatively consistent across all six states, remaining around 21–23%

In [28]:
print(rawsales["Order Date"])

0       2023-06-27
1       2024-12-27
2       2021-07-25
3       2023-06-27
4       2024-12-27
           ...    
1189    2024-07-31
1190    2020-06-02
1191    2022-12-15
1192    2020-08-07
1193    2024-10-26
Name: Order Date, Length: 1194, dtype: object


In [29]:
rawsales.rename(columns={"order date": "Order_Date"}, inplace=True)

In [30]:
rawsales.columns

Index(['Order ID', 'Amount', 'Profit', 'Quantity', 'Category', 'Sub-Category',
       'PaymentMode', 'Order Date', 'CustomerName', 'State', 'City',
       'Year-Month'],
      dtype='object')

In [31]:
rawsales = rawsales.rename(columns={"Order ID": "Order_ID", "Order Date": "Order_Date"})

In [32]:
rawsales.columns

Index(['Order_ID', 'Amount', 'Profit', 'Quantity', 'Category', 'Sub-Category',
       'PaymentMode', 'Order_Date', 'CustomerName', 'State', 'City',
       'Year-Month'],
      dtype='object')

In [36]:
rawsales["Order_Date"].head(10)

0    2023-06-27
1    2024-12-27
2    2021-07-25
3    2023-06-27
4    2024-12-27
5    2021-07-25
6    2024-05-11
7    2021-10-09
8    2022-11-18
9    2022-11-18
Name: Order_Date, dtype: object

In [38]:
rawsales["Order_Date"] = pd.to_datetime(rawsales["Order_Date"])

In [39]:
rawsales["Order_Date"].dtype

dtype('<M8[ns]')

In [41]:
rawsales["Order_Date"].dt.to_period("M").astype(str)

0       2023-06
1       2024-12
2       2021-07
3       2023-06
4       2024-12
         ...   
1189    2024-07
1190    2020-06
1191    2022-12
1192    2020-08
1193    2024-10
Name: Order_Date, Length: 1194, dtype: object

In [44]:
ww = rawsales["Order_Date"].dt.to_period("M").astype(str) == rawsales["Year-Month"]

In [48]:
ww.value_counts()

True    1194
Name: count, dtype: int64

In [51]:
rawsales.columns

Index(['Order_ID', 'Amount', 'Profit', 'Quantity', 'Category', 'Sub-Category',
       'PaymentMode', 'Order_Date', 'CustomerName', 'State', 'City'],
      dtype='object')

In [53]:
rawsales["Order_ID"].nunique()

547

In [54]:
rawsales["Order_ID"].value_counts()

Order_ID
B-26489    12
B-26032    12
B-25465    12
B-26279    10
B-25715     9
           ..
B-26545     1
B-26603     1
B-25945     1
B-26565     1
B-25068     1
Name: count, Length: 547, dtype: int64